src/idlcxx/src/streamers.c

```diff
@@ -169,27 +178,45 @@ unroll_sequence(const idl_pstate_t* pstate,
     return IDL_RETCODE_NO_MEMORY;
 
   idl_retcode_t ret = IDL_RETCODE_OK;
+  char* new_read_accessor = NULL;
+  if (maximum)
+  {
+    if (idl_asprintf(&new_read_accessor, "temp_%u", depth) < 0)
+    {
+      ret = IDL_RETCODE_NO_MEMORY;
+      goto done;
```

Comment:
Early return please, same for the rest of the code.

Reply:
will do